cns/restserver/v2/server_test.go

```diff
@@ -0,0 +1,101 @@
+package v2
+
+import (
+	"net"
+	"net/url"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/Azure/azure-container-networking/cns/common"
+	"github.com/Azure/azure-container-networking/cns/fakes"
+	"github.com/Azure/azure-container-networking/cns/logger"
+	"github.com/Azure/azure-container-networking/cns/restserver"
+	acncommon "github.com/Azure/azure-container-networking/common"
+	"github.com/pkg/errors"
+)
+
+// TestStartServices will test three scenarios:
+// 1. when customer provides -p option, make sure local server is running and server is using this port
+// 2. when customer provides -c option, local server will this -c URL
+
+func TestStartServerWithCNSPort(t *testing.T) {
+	var err error
+
+	logger.InitLogger("testlogs", 0, 0, "./")
+	cnsPort := "8000"
+
+	// Create the service with -p 8000
+	if err = startService(cnsPort, ""); err != nil {
+		t.Errorf("Failed to connect to CNS Service on expected port:%s. Error: %v", cnsPort, err)
+		os.Exit(1)
```

Comment:
You should use Fatalf here instead and omit `os.Exit`. It shuts down the test less violently.

```suggestion
		t.Fatalf("Failed to connect to CNS Service on expected port:%s. Error: %v", cnsPort, err)
```